### ISSUE TITLE
Character detection failure

### ISSUE THREAD

**Issue by CoffeeFlux** [TypesettingTools/Aegisub#48]
Attempting the load the following file will error with invalid characters, despite that it appears to be proper UTF-8: https://sharex.thevacuumof.space/2019/03/test02.ass

**Comment by davste0816** [comment]
It's due to the Project Garbage being out of bounds. Try this: 
[test.zip](https://github.com/TypesettingTools/Aegisub/files/2969584/test.zip)

**Comment by CoffeeFlux** [comment]
Oh huh, that may be a different issue than what's affecting the full file, though either way that's pretty poor behavior - it shouldn't either fail to load or improperly convert characters as a result of the project garbage.

**Comment by CoffeeFlux** [comment]
No, I tested and out of bounds project garbage is not what's causing it. Removing that section seems to avoid the character conversion problem, but you can do the same by removing various other lines as well.

**Comment by davste0816** [comment]
Try removing just the lines:
```
Audio File: hsg01-1080V (1).mkv
Video File: hsg01-1080V (1).mkv
```
works. It's something here maybe?

**Comment by davste0816** [comment]
In fact, one of these numbers `01108` is coming up as non-UTF8.

**Comment by CoffeeFlux** [comment]
Those lines seem to be fine - again, a lot of different random lines can be removed here. It's some strange iconv issue it seems, and I'm not sure further speculation is very useful; I need to just step through it in a debugger.

With that said, it seems like the issue here may be different from the original file, so I'm uploading that as well: https://thevacuumof.space/hi_score_girl_01_broken.ass

If they're demonstrated to be different, I'll open a separate issue for the full file.

**Comment by FichteFoll** [comment]
As a summary from what we assessed on Discord, this is caused by:

1. A missing DOM that is required for some reason (iconv bug?)
2. A hard file size limit of 100MB coded somewhere in the Aegisub code base
